docs/design/autotuing-design.md

```diff
@@ -0,0 +1,269 @@
+ provide optimization strategies based on the current training status. This
+ includes suggesting new batch sizes and learning rates to maximize GPU
+ resource utilization.
+
+ **Dynamic Batch Size Configuration:** Leveraging the
+ collected and analyzed performance data, we'll devise an algorithm that
+ maximizes GPU resource utilization during task submissions while preventing
+ Out-of-Memory (OOM) errors.
+
+ **Dynamic Learning Rate Configuration:** Even as
+ we focus on boosting training speed, maintaining precision despite potential
+ changes in batch size remains paramount. Thus, we will concentrate on
+ configuring and updating learning rates for training tasks, designing
+ corresponding algorithms.
+
+## Design
+
+### Process and Component Design
+
+#### Key Components
+
+`GPU Stats Collector`: A subcomponent of resource monitor to add gpu information collection.
+
+`Hyper-parameter Tuner`: The main generative component of the optimization algorithm is generated in.
+
+`JobMaster` by optimizing the configuration based on the training information of the nodes.
+
+`Elastic Dataloader`: The main execution component of dynamic configuration dynamically
+adjusts the batch size of the training process based on the generated
+hyper-reference configuration.
+
+#### Main Workflow
+
+<div align="center">
+<img src="../figures/dlrover-auto-tuning.png" alt="Autotuning" width="1000">
+</div>
+As shown in the figure, introspective hyper-parameter tuning in **DLRover**
+contains the following steps:
+
+- The `GPU Stats Collector` within `ElasticAgent.ResourceMonitor` gathers
+  training resource information to create  a `ResourceStat`[used_mem,
+  current_cpu, gpu_stats].
+- The `Master_Client` reports `ResourceStat` to `gRPC Server`
+- The `Master.Servicer`receives the `ResourceStat` through a  `gRPC Request`
+  and update `NodeResource`
+- `Master.Node.JobManager` obtains the  `ResourceStat` from `NodeResource` and
+  and enters a waiting state for `HyperParameterTuner`.
+- `HyperParameterTuner` generates a new `HyperParameterConfigPlan` and
+  dispatches a query to `ElasticAgent`
+- `ElasticAgent.ConfigGenerator` updates the `Config File`.
+- The `Trainer` in `Worker` checks the `Config File` timely and  adjust the
+  hpyer-parameters based on its contents. For example, the `Elastic Dataloader`
+  fine-tunes the batch size in accordance with the new configuration.
+
+### Data Structure Design
+
+#### GPUMetric
+
+```python
+@dataclass
+class GPUMetric:
+    """GPU performance metrics for Container."""
+    
+    index: int
+    """The index of the GPU."""
+    
+    total_memory_mb: int
+    """The total memory of the GPU in megabytes (MB)."""
+    
+    used_memory_mb: int
+    """The amount of used memory on the GPU in megabytes (MB)."""
+    
+    gpu_utilization: float
+    """The GPU utilization as a percentage."""
+```
+
+#### HyperParas
+
+```python
+@dataclass
+class HyperParas:
```

Comment:
We usually use params for short of parameters.

Reply:
Apologize for my oversight. All of these were fixed by the latest commits.